### PR TITLE
Add django support

### PR DESCRIPTION
**Pull request by prodion23**
## Description
Adds django support + tests

This also includes the perf improvements PR(wanted to go ahead and tackle the required changes + merge conflicts) once that one is merged this pr will be smaller

### CODE REVIEW

File: src/hypertrace/agent/instrumentation/django/__init__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        DjangoInstrumentor().instrument(request_hook=self.request_hook,
+                                        response_hook=self.response_hook)
+
+    def request_hook(self, span: Span, request):
+        """django request hook before request is processed by app"""
+        try:
+            body = request.body
+            span.update_name(f"{request.method} {span.name}")
+            self.generic_request_handler(request.headers, body, span)
+            full_url = request.build_absolute_uri()
```
Comment (jcchavezs) [inline src/hypertrace/agent/instrumentation/django/__init__.py:30]: Does this trim query param? That is usually sensitive.

File: tests/django/agent-config.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+service_name: pythonagent
+reporting:
+  trace_reporter_type: ZIPKIN
+  endpoint: http://localhost:9411/api/v2/spans
+  secure: true
```
Comment (jcchavezs) [inline tests/django/agent-config.yaml:5]: Just curious if secure is needed for a localhost http connection.

File: tests/django/tox.ini [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+whitelist_externals =
+    pytest
+    sleep
+    docker-compose
+
+setenv =
+  PYTHONPATH=../../src
+  HT_CONFIG_FILE=./agent-config.yaml
+  HT_SERVICE_NAME=python_agent_test1
+#  HT_REPORTING_TRACE_REPORTER_TYPE=zipkin
```
Comment (jcchavezs) [inline tests/django/tox.ini:21]: Shall we drop comments?

**Review by jcchavezs — APPROVED** [review]
LGTM

File: src/hypertrace/agent/instrumentation/django/__init__.py (hunk 1 above)
Comment (prodion23) [reply src/hypertrace/agent/instrumentation/django/__init__.py:30]: No, just scheme + host + path

**Review by davexroth — APPROVED** [review]

**Pull request merged**